jaxley/modules/base.py

```diff
@@ -227,39 +286,37 @@ def reindex_a_by_b(
             return df
 
         index_names = ["cell_index", "branch_index", "comp_index"]  # order is important
-        for obj, prefix in zip(
-            [self.nodes, self.edges, self.edges], ["", "pre_", "post_"]
-        ):
-            global_idx_cols = [f"global_{prefix}{name}" for name in index_names]
-            local_idx_cols = [f"local_{prefix}{name}" for name in index_names]
-            idcs = obj[global_idx_cols]
+        global_idx_cols = [f"global_{name}" for name in index_names]
+        local_idx_cols = [f"local_{name}" for name in index_names]
+        idcs = self.nodes[global_idx_cols]
 
-            idcs = reindex_a_by_b(idcs, global_idx_cols[0])
-            idcs = reindex_a_by_b(idcs, global_idx_cols[1], global_idx_cols[0])
-            idcs = reindex_a_by_b(idcs, global_idx_cols[2], global_idx_cols[:2])
-            idcs.columns = [col.replace("global", "local") for col in global_idx_cols]
-            obj[local_idx_cols] = idcs[local_idx_cols].astype(int)
+        # update local indices of nodes
+        idcs = reindex_a_by_b(idcs, global_idx_cols[0])
+        idcs = reindex_a_by_b(idcs, global_idx_cols[1], global_idx_cols[0])
+        idcs = reindex_a_by_b(idcs, global_idx_cols[2], global_idx_cols[:2])
+        idcs.columns = [col.replace("global", "local") for col in global_idx_cols]
+        self.nodes[local_idx_cols] = idcs[local_idx_cols].astype(int)
 
         # move indices to the front of the dataframe; move controlled_by_param to the end
+        # move indices of current scope to the front and the others to the back
+        not_scope = "global" if self._scope == "local" else "local"
+        self.nodes = reorder_cols(
```

Comment:
If scope is local, local_inds will be shown at the front, globals at the back and vice versa.